client/core/trade.go

```diff
@@ -1267,6 +1299,83 @@ func (t *trackedTrade) finalizeRedeemAction(match *matchTracker, coinID []byte)
 	return errs.ifAny()
 }
 
+func (t *trackedTrade) beginFindRedemption(match *matchTracker) {
+	swapCoinID := dex.Bytes(match.MetaData.Proof.TakerSwap)
+	contract := dex.Bytes(match.MetaData.Proof.CounterScript)
+	fromID := t.wallets.fromAsset.ID
+	resultChan, err := t.wallets.fromWallet.FindRedemption(swapCoinID, contract)
```

Comment:
> The swapCoinID is enough to get the contract output's pkScript, which gives the script hash (ExtractScriptHash)

That basically does it and was the approach but the contract output's pkScript is not included in the result of `node/wallet.GetTransaction(hash)` which the dcr/btc ExchangeWallet's use to get the transaction information. The `GetTransaction` result does include the tx hex which I initially decoded to a msgTx and retrieved the contract output from; but the dcr and btc asset tests failed because the mocked `GetTransaction` results do not include a valid hex.

Thinking about it again, I agree that the contract __need not__ be an input to the FindRedemption method, so I've reverted to decoding the tx hex and making adjustments to the tests to include the hex in the mocked `GetTransaction` result.